dali_tf_plugin/dali_dataset_op.cc

```diff
@@ -183,6 +226,23 @@ class DALIDatasetOp::Dataset::Iterator : public DatasetIterator<Dataset> {
         enable_memory_stats_(enable_memory_stats) {}
 
   Status Initialize(IteratorContext *context) override {
+    // Based on ZipDataset
+    mutex_lock l(mu_);
+    iterator_state_ = InputState::in_progress;
+    if (dataset()->HasInputs()) {
+      input_impls_.resize(dataset()->NumInputs());
+      for (size_t i = 0; i < input_impls_.size(); ++i) {
+#if TF_MAJOR_VERSION == 2 && TF_MINOR_VERSION >= 3
```

Comment:
How about making it a feature only for newer TF versions and just disable the whole thing for TF < 2.3

Reply:
I think if this was like a major overhaul of the whole interface than we probably shouldn't care, but this is just passing one additional pointer and not much changes for us. And we care about nvidia/tensorflow